frontend/src/app/homepage-offer/homepage-offer.component.ts

```diff
@@ -12,6 +13,8 @@ import { OrganizationService } from '../organization/organization.service';
   changeDetection: ChangeDetectionStrategy.OnPush,
```

Comment:
This line is blocking the change detection mechanism, so Angular thinks that this component will not change. Please comment it for now - your offer changes should work after commenting. We can leave it commented for further investigation, at it is actually not a bad idea to push changes manually here, but I don't have the solution for that yet.

Reply:
I did not run this code and I don't know what's not working, but `OnPush` is usually a good idea.
It's nowhere near to require pushing changes "manually", it just requires using immutable objects as inputs. Apparently some of our input objects are mutated (which I'm afraid was introduced by the way offer image is set and maybe elsewhere, too), which at best should be avoided.

Reply:
As it turned out, it was not the problem with ChangeDetectionStrategy.OnPush, but with backend/frontend communication on image save, which threw an error, hence no changes to the image. 
We can leave the ChangeDetectionStrategy.OnPush here.